### PR TITLE
Add ArrayToYieldDataProviderRector

### PR DESCRIPTION
**Pull request by TomasVotruba**
### Before

```php
    /**
     * @return mixed[]
     */
    public function provideData(): array
    {
        return [
            ['item']
        ];
    }
```

### After

```php
    /**
     */
    public function provideData(): \Iterator
    {
        (yield ['item']);
    }
```

### CODE REVIEW

File: src/Rector/Architecture/PHPUnit/ArrayToYieldDataProviderRector.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+        // 3. remove doc block
+        $this->docBlockAnalyzer->removeAnnotationFromNode($classMethodNode, 'return', '');
+
+        return $classMethodNode;
+    }
+
+    private function isInProvideMethod(ClassMethod $classMethodNode): bool
+    {
+        return (bool) Strings::match($classMethodNode->name->toString(), '#^provide*#');
```
Comment (carusogabriel) [inline src/Rector/Architecture/PHPUnit/ArrayToYieldDataProviderRector.php]: Doesn't it already return `bool`?

File: tests/Rector/Architecture/PHPUnit/Correct/correct.php.inc [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?php declare(strict_types=1);
+
+namespace App\Tests;
+
+use PHPUnit\Framework\TestCase;
+
+final class MyTestCsae extends TestCase
+{
+    /**
+     */
```
Comment (carusogabriel) [inline tests/Rector/Architecture/PHPUnit/Correct/correct.php.inc]: Can't we remove them complete?

Comment (TomasVotruba) [reply tests/Rector/Architecture/PHPUnit/Correct/correct.php.inc]: That's job of coding standard

File: src/Rector/Architecture/PHPUnit/ArrayToYieldDataProviderRector.php (hunk 1 above)
Comment (TomasVotruba) [reply src/Rector/Architecture/PHPUnit/ArrayToYieldDataProviderRector.php]: It return matched result or `false`

File: tests/Rector/Architecture/PHPUnit/Correct/correct.php.inc [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    public function provideData(): \Iterator
+    {
+        (yield ['item']);
```
[the remark this answers is not in the document]
Comment (nikic) [reply tests/Rector/Architecture/PHPUnit/Correct/correct.php.inc]: I'd suggest extending the pretty printer and drop the parens if the parent node is a Stmt\Expression at least.

[the remark this answers is not in the document]
Comment (TomasVotruba) [reply tests/Rector/Architecture/PHPUnit/Correct/correct.php.inc]: Thank you, I'll try it

[the remark this answers is not in the document]
Comment (TomasVotruba) [reply tests/Rector/Architecture/PHPUnit/Correct/correct.php.inc]: https://github.com/rectorphp/rector/pull/421
Like this?

[the remark this answers is not in the document]
Comment (nikic) [reply tests/Rector/Architecture/PHPUnit/Correct/correct.php.inc]: Yeah, like that.

**Pull request merged**